cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -50,7 +51,7 @@
  * @author Andrew Tillinghast
  * @since 3.5
  */
-public final class EhCacheTicketRegistry extends AbstractDistributedTicketRegistry implements InitializingBean {
+public final class EhCacheTicketRegistry extends AbstractCrypticTicketRegistry implements InitializingBean {

```

Comment:
> The ticketRegistry is defined in the ticketRegistry.xml file, so that's the only place you need to change something: rename the default one and create this new one.

No this is not true. It's not enough to change it there. Other places need to change as well that reference that bean. For example CASImpl expects a `ticketRegistry`. Extensions expect that bean id. In a MFA context, everything pretty much references that bean id. So can not change it. 

If we do the wrapper option it turns into:

```xml
<bean id="crypticRegistry>
<property name="innerRegistry" ref="ticketRegistry" />
</bean>
```

Then, every place that references the `ticketRegistry` must be changed to point to `crypticRegistry`. That is a lot of work both for us, and the adopter in a deployment.

The above model also makes no guarantee from an API perspective, what the inner ticket registry may be. You are just passing a TicketRegistry. Because a ticket registry that is not cache-based and does not do serialization cannot support encryption. You can't protect what gets passed to you without API changes, so you only solution is to throw exceptions when things fail, and then document "X is supported, but Y isnt"

Because of that, you have to maintain a separate configuration block and ask people to configure wrappers and inners accordingly and go through their configuration to make sure everything they have defined as ticketRegistry is now updated to support that. Also not an option. 

And if you're going to make API changes to protect against what's passed to you, you might as well do it correctly. Wrappers work better if you're writing extensions. Since we have access to the API directly, might as well do it directly. When you enable Ehcache or Hazelcast ticket registries, you should *not* have to make a decision and then understand whether your choice supports encryption or not based on docs and user forums. You automatically get that support. Just have to enabled it. You also should not have to define wrappers to communicate the idea that you're trying to intercept something. If you can do it directly, why intercept it?